src/components/user-avatar.component.jsx

```diff
@@ -2,10 +2,24 @@ import * as React from 'react';
 
 import Avatar from 'material-ui/Avatar';
 import SocialPerson from 'material-ui/svg-icons/social/person';
+import IconMenu from 'material-ui/IconMenu';
+import MenuItem from 'material-ui/MenuItem';
+import IconButton from 'material-ui/IconButton';
 
-const UserAvatar = ({ user, onTouchTap, style }) => (
-    user.vkPhotoUrl ? <Avatar src={user.vkPhotoUrl} style={style} onTouchTap={onTouchTap} />
-        : <Avatar icon={<SocialPerson />} style={style} onTouchTap={onTouchTap} />
+const UserAvatar = ({ user, onSettingsTap, style }) => (
+      <IconMenu
+          style={{marginRight: '2px'}}
+          iconButtonElement={
+              <IconButton style={{padding: '0px'}}>{
+                  user.vkPhotoUrl ? <Avatar src={user.vkPhotoUrl} style={style} /> :
+                  <Avatar icon={<SocialPerson />} style={style} />}
+              </IconButton>}
+          anchorOrigin={{horizontal: 'left', vertical: 'bottom'}}
+          targetOrigin={{horizontal: 'middle', vertical: 'top'}}
+          >
+          <MenuItem primaryText='Settings' onTouchTap={onSettingsTap}/>
+          <MenuItem primaryText='Sign out' href='/logout'/>
```

Comment:
In the end, MenuItem is a <a>. The prop gets passed down which is why it works. 
On another note, I used href instead of Link because link is used for internal routing, where href takes us to an external route, which then redirects us back to homepage with expired cookie. Basically, I applied the same approach that was done for login. I can change it to link and create a component that does an api call to logout and clear the account.